src/elm/list.ts

```diff
@@ -5,19 +5,19 @@ import { equals } from '../util/comparison';
 import { Context } from '../runtime/context';
 
 class List extends Expression {
-  elements: any[];
+  elements: Expression[];
 
   constructor(json: any) {
     super(json);
-    this.elements = build(json.element) || [];
+    this.elements = (build(json.element) as Expression[]) || [];
   }
 
   get isList() {
     return true;
   }
 
   async exec(ctx: Context) {
-    return this.elements.map(item => item.execute(ctx));
+    return await Promise.all(this.elements.map(async item => await item.execute(ctx)));
```

Comment:
whoops. thanks

Reply:
Actually looking at this again I wonder if this would work...
```suggestion
    return await Promise.all(this.elements.map(item => item.execute(ctx)));
```

Reply:
I think your suggestion there will work.. since `item.execute` returns a promise, we are passing in an array of promises to `Promise.all`, so it should be kosher. I tested a small example in a repl and it agreed